src/Ledger/Utxo/Properties.lagda

```diff
@@ -91,39 +104,42 @@ consumedCoinEquality : ∀ {pp} → coin (mint tx) ≡ 0
-  coin (balance (utxo ∣ txins tx) +ᵛ mint tx +ᵛ inject (depositRefunds pp utxoState tx))
+  coin (balance (utxo ∣ txins tx) + mint tx + inject (depositRefunds pp utxoState tx))
     ≡⟨ ∙-homo-Coin _ _ ⟩
-  coin (balance (utxo ∣ txins tx) +ᵛ mint tx) + coin (inject $ depositRefunds pp utxoState tx)
-    ≡⟨ cong (coin (balance (utxo ∣ txins tx) +ᵛ mint tx) +_) (property _) ⟩
-  coin (balance (utxo ∣ txins tx) +ᵛ mint tx) + depositRefunds pp utxoState tx
-    ≡⟨ cong! (∙-homo-Coin _ _) ⟩
+  coin (balance (utxo ∣ txins tx) + mint tx) + coin (inject $ depositRefunds pp utxoState tx)
+    ≡⟨ cong (coin (balance (utxo ∣ txins tx) + mint tx) +_) (property _) ⟩
+  coin (balance (utxo ∣ txins tx) + mint tx) + depositRefunds pp utxoState tx
+    ≡⟨ cong (_+ depositRefunds pp utxoState tx) (∙-homo-Coin _ _) ⟩
   coin (balance (utxo ∣ txins tx)) + coin (mint tx) + depositRefunds pp utxoState tx
     ≡⟨ cong (λ x → cbalance (utxo ∣ txins tx) + x + depositRefunds pp utxoState tx) h ⟩
   cbalance (utxo ∣ txins tx) + 0 + depositRefunds pp utxoState tx
-    ≡⟨ cong! (+-identityʳ (cbalance (utxo ∣ txins tx))) ⟩
+    ≡⟨ cong (_+ depositRefunds pp utxoState tx) (+-identityʳ (cbalance (utxo ∣ txins tx))) ⟩
   cbalance (utxo ∣ txins tx) + depositRefunds pp utxoState tx                        ∎
 
-producedCoinEquality : ∀ {pp} → coin (produced pp utxoState tx)
-                              ≡ cbalance (outs tx) + (txfee tx) + newDeposits pp utxoState tx + txdonation tx
+
+producedCoinEquality : ∀ {pp}
+  → coin (produced pp utxoState tx) ≡ cbalance (outs tx) + txfee tx + newDeposits pp utxoState tx + txdonation tx
+
 producedCoinEquality {utxoState} {tx} {pp} = begin
-  coin (balance (outs tx) +ᵛ inject (txfee tx) +ᵛ inject (newDeposits pp utxoState tx) +ᵛ inject (txdonation tx))
-    ≡⟨ ∙-homo-Coin _ _ ⟩
-  coin (balance (outs tx) +ᵛ inject (txfee tx) +ᵛ inject (newDeposits pp utxoState tx)) + coin (inject (txdonation tx))
-    ≡⟨ cong (_+ coin (inject (txdonation tx))) (begin
-      coin (balance (outs tx) +ᵛ inject (txfee tx) +ᵛ inject (newDeposits pp utxoState tx))
-        ≡⟨ ∙-homo-Coin _ _ ⟩
-      coin (balance (outs tx) +ᵛ inject (txfee tx)) + coin (inject (newDeposits pp utxoState tx))
-        ≡⟨ cong! (property _) ⟩
-      coin (balance (outs tx) +ᵛ inject (txfee tx)) + newDeposits pp utxoState tx
-        ≡⟨ cong! (∙-homo-Coin _ _) ⟩
-      coin (balance (outs tx)) + coin (inject (txfee tx)) + newDeposits pp utxoState tx
-        ≡⟨ cong (λ x → cbalance (outs tx) + x + newDeposits pp utxoState tx) (property (txfee tx)) ⟩
-      cbalance (outs tx) + (txfee tx) + newDeposits pp utxoState tx                     ∎
-    )⟩
-  cbalance (outs tx) + (txfee tx) + newDeposits pp utxoState tx + coin (inject (txdonation tx))
-    ≡⟨ cong (cbalance (outs tx) + (txfee tx) + newDeposits pp utxoState tx +_) (property _) ⟩
-  cbalance (outs tx) + (txfee tx) + newDeposits pp utxoState tx + txdonation tx
-    ∎
```

Comment:
I think we should try to avoid nesting equational reasoning proofs.  The re-written version is easier to parse and understand (imho).

Reply:
reverted ∵ change requests